src/node_webstorage.cc

```diff
@@ -0,0 +1,706 @@
+  node::THROW_ERR_INVALID_STATE((env), sqlite3_errstr((r)))
+
+#define CHECK_ERROR_OR_THROW(env, expr, expected, ret)                         \
+  do {                                                                         \
+    int r_ = (expr);                                                           \
+    if (r_ != (expected)) {                                                    \
+      THROW_SQLITE_ERROR((env), r_);                                           \
+      return (ret);                                                            \
+    }                                                                          \
+  } while (0)
+
+static void ThrowQuotaExceededException(Local<Context> context) {
+  Isolate* isolate = context->GetIsolate();
+  auto dom_exception_str = FIXED_ONE_BYTE_STRING(isolate, "DOMException");
+  auto err_name = FIXED_ONE_BYTE_STRING(isolate, "QuotaExceededError");
+  auto err_message =
+      FIXED_ONE_BYTE_STRING(isolate, "Setting the value exceeded the quota");
+  Local<Object> per_context_bindings;
+  Local<Value> domexception_ctor_val;
+  if (!GetPerContextExports(context).ToLocal(&per_context_bindings) ||
+      !per_context_bindings->Get(context, dom_exception_str)
+           .ToLocal(&domexception_ctor_val)) {
+    return;
+  }
+  CHECK(domexception_ctor_val->IsFunction());
+  Local<Function> domexception_ctor = domexception_ctor_val.As<Function>();
+  Local<Value> argv[] = {err_message, err_name};
+  Local<Value> exception;
+
+  if (!domexception_ctor->NewInstance(context, arraysize(argv), argv)
+           .ToLocal(&exception)) {
+    return;
+  }
+
+  isolate->ThrowException(exception);
+}
+
+Storage::Storage(Environment* env, Local<Object> object, Local<String> location)
+    : BaseObject(env, object) {
+  MakeWeak();
+  node::Utf8Value utf8_location(env->isolate(), location);
+  symbols_.Reset(env->isolate(), Map::New(env->isolate()));
+  db_ = nullptr;
+  location_ = utf8_location.ToString();
+}
+
+Storage::~Storage() {
+  db_ = nullptr;
+}
+
+void Storage::MemoryInfo(MemoryTracker* tracker) const {
+  tracker->TrackField("symbols", symbols_);
+  tracker->TrackField("location", location_);
+}
+
+bool Storage::Open() {
+  static const int kCurrentSchemaVersion = 1;
+  static const char get_schema_version_sql[] =
+      "SELECT schema_version FROM nodejs_webstorage_state";
+  static const char init_sql_v0[] =
+      "PRAGMA encoding = 'UTF-16le';"
+      "PRAGMA busy_timeout = 3000;"
+      "PRAGMA journal_mode = WAL;"
+      "PRAGMA synchronous = NORMAL;"
+      "PRAGMA temp_store = memory;"
+      "PRAGMA optimize;"
+      ""
+      "CREATE TABLE IF NOT EXISTS nodejs_webstorage("
+      "  key BLOB NOT NULL,"
+      "  value BLOB NOT NULL,"
+      "  PRIMARY KEY(key)"
+      ") STRICT;"
+      ""
+      "CREATE TABLE IF NOT EXISTS nodejs_webstorage_state("
+      // max_size is 10MB. This can be made configurable in the future.
+      "  max_size INTEGER NOT NULL DEFAULT 10485760,"
```

Comment:
Does this only apply to `sessionStoage` or to `localStorage` too? If yes, in the docs, you only mentioned the limit for `sessionStorage`.